scl/tags.ts

```diff
@@ -0,0 +1,847 @@
+  SettingGroups: {
+    parents: ['Services'],
+    children: ['SGEdit', 'ConfSG'],
+  },
+  SGEdit: {
+    parents: ['SettingGroups'],
+    children: [],
+  },
+  SmpRate: {
+    parents: ['SMVSettings'],
+    children: [],
+  },
+  SMV: {
+    parents: ['ConnectedAP'],
+    children: [...tControlBlockSequence],
+  },
+  SmvOpts: {
+    parents: ['SampledValueControl'],
+    children: [],
+  },
+  SMVsc: {
+    parents: ['Services'],
+    children: [],
+  },
+  SMVSecurity: {
+    parents: ['AccessPoint'],
+    children: [...tNamingSequence, 'Subject', 'IssuerName'],
+  },
+  SMVSettings: {
+    parents: ['Services'],
+    children: ['SmpRate', 'SamplesPerSec', 'SecPerSamples', 'McSecurity'],
+  },
+  SubEquipment: {
+    parents: [
+      'TapChanger',
+      'PowerTransformer',
+      'ConductingEquipment',
+      'TransformerWinding',
+      ...tAbstractConductingEquipment,
+    ],
+    children: [...tPowerSystemResourceSequence, 'EqFunction'],
+  },
+  SubFunction: {
+    parents: ['SubFunction', 'Function'],
+    children: [
+      ...tPowerSystemResourceSequence,
+      'GeneralEquipment',
+      'ConductingEquipment',
+      'SubFunction',
+    ],
+  },
+  SubNetwork: {
+    parents: ['Communication'],
+    children: [...tNamingSequence, 'BitRate', 'ConnectedAP'],
+  },
+  Subject: {
+    parents: ['GOOSESecurity', 'SMVSecurity'],
+    children: [],
+  },
+  Substation: {
+    parents: ['SCL'],
+    children: [...tEquipmentContainerSequence, 'VoltageLevel', 'Function'],
+  },
+  SupSubscription: {
+    parents: ['Services'],
+    children: [],
+  },
+  TapChanger: {
+    parents: ['TransformerWinding'],
+    children: [...tPowerSystemResourceSequence, 'SubEquipment', 'EqFunction'],
+  },
+  Terminal: {
+    parents: [...tEquipment],
+    children: [...tUnNamingSequence],
+  },
+  Text: {
+    parents: sCLTags.filter(tag => tag !== 'Text' && tag !== 'Private'),
+    children: [],
+  },
+  TimerActivatedControl: {
+    parents: ['Services'],
+    children: [],
+  },
+  TimeSyncProt: {
+    parents: ['Services', 'ClientServices'],
+    children: [],
+  },
+  TransformerWinding: {
+    parents: ['PowerTransformer'],
+    children: [
+      ...tAbstractConductingEquipmentSequence,
+      'TapChanger',
+      'NeutralPoint',
+      'EqFunction',
+      'SubEquipment',
+    ],
+  },
+  TrgOps: {
+    parents: ['ReportControl'],
+    children: [],
+  },
+  Val: {
+    parents: ['DAI', 'DA', 'BDA'],
+    children: [],
+  },
+  ValueHandling: {
+    parents: ['Services'],
+    children: [],
+  },
+  Voltage: {
+    parents: ['VoltageLevel'],
+    children: [],
+  },
+  VoltageLevel: {
+    parents: ['Substation'],
+    children: [...tEquipmentContainerSequence, 'Voltage', 'Bay', 'Function'],
+  },
+};
+
+const tagSet = new Set<string>(sCLTags);
+
+function isSCLTag(tag: string): tag is SCLTag {
+  return tagSet.has(tag);
+}
+
+/** @returns parent `tagName` s for SCL (2007B4) element tag  */
+export function parentTags(tagName: string): string[] {
```

Comment:
I think we could possibly return an `SCLTag[]` from these two functions and export that type and the `isSCLTag` guard.

Reply:
I am not sure whether we can export `SCLTag` as type. It is based in `2007B4`. I am not sure if this can be done based on the schema version of the project SCL file.